nav2_route/src/goal_intent_extractor.cpp

```diff
@@ -53,6 +60,15 @@ void GoalIntentExtractor::configure(
   nav2_util::declare_parameter_if_not_declared(
     node, "min_dist_from_start", rclcpp::ParameterValue(0.10));
   min_dist_from_start_ = static_cast<float>(node->get_parameter("min_dist_from_start").as_double());
+
+  std::string global_costmap_topic;
+  nav2_util::declare_parameter_if_not_declared(
+    node, "global_costmap_topic", rclcpp::ParameterValue(std::string("global_costmap/costmap_raw")));
+  node->get_parameter("global_costmap_topic", global_costmap_topic);
+
+  costmap_sub_ = std::make_unique<nav2_costmap_2d::CostmapSubscriber>(node, global_costmap_topic);
+  bfs_ = std::make_unique<BreadthFirstSearch>();
```

Comment:
BFS costmap checking should be a parameterized feature. It might not be viable for many real situations.